### ISSUE TITLE
multiple build pods being spawned when importing a quickstart or starting pipelines

### ISSUE THREAD

**Issue by mikecirioli** [jenkins-x/jx#2515]
### Summary 
When importing a new quickstart (does not seem to matter what type), I am seeing 2 build pods being spawned.  One of these will succeed, and one will fail due to tagging error.  Furthermore, subsequent pipeline builds (started after the initial builds have completed using `jx start pipeline`) will also result in duplicate build pods.  To make matters worse, at some point starting a new build I see previous builds spontaneously create new pods.  ex.  `jx start pipeline` kicks off build #5, and `jx get build pods` shows that build #3 suddenly grew from 2 to 4 pods.

### Steps to reproduce the behavior
`jx create spring` or `jx create quickstart -> golang/http`

I consistently see 2 build pods started, 2 duplicate entires in `jx get build logs`, etc.  

Starting subsequent builds also consistently triggers multiple build pods.

```
k8s:(core-dev-1|racermud-shared-dev|jx)
[mcirioli@lagavulin /export/cloudbees/git/jx_playground]$ jx start pipeline
? Which pipeline do you want to start:  cb-kubecd/mike-prow1/master
k8s:(core-dev-1|racermud-shared-dev|jx)
[mcirioli@lagavulin /export/cloudbees/git/jx_playground]$ jx get build logs
? Which build do you want to view the logs of?:   [Use arrows to move, type to filter]
> cb-kubecd/mike-prow1/master #2
  cb-kubecd/mike-prow1/master #2
error: interrupt
k8s:(core-dev-1|racermud-shared-dev|jx)
[mcirioli@lagavulin /export/cloudbees/git/jx_playground]$ jx get build pods
OWNER     REPOSITORY BRANCH BUILD AGE STATUS  STEP 1 IMAGE                  POD                                        GIT URL
cb-kubecd mike-prow1 master 2         Pending jenkinsxio/jenkins-go:0.1.156 cede1634-ffbe-11e8-a756-6a0003553450-95p2f https://github.com/cb-kubecd/mike-prow1.git
cb-kubecd mike-prow1 master 2         Pending jenkinsxio/jenkins-go:0.1.156 cede1634-ffbe-11e8-a756-6a0003553450-f548c https://github.com/cb-kubecd/mike-prow1.git
```
### Jx version

The output of `jx version` is:

```
Failed to find helm installs: failed to run 'helm list' command in directory '', output: 'Error: incompatible versions client[v2.12.0] server[v2.11.0]': exit status 1
NAME               VERSION
jx                 1.3.661
Kubernetes cluster v1.9.7-gke.11
kubectl            v1.12.2
helm client        v2.12.0+gd325d2a
helm server        v2.11.0+g2e55dbe
git                git version 2.16.0
Operating System   Mac OS X 10.14.1 build 18B75
```

### Kubernetes cluster

What kind of Kubernetes cluster are you using & how did you create it?
google cloud, created via `jx install`

### Expected behavior
one build per action

### Actual behavior
see description

**Comment by mikecirioli** [comment]
Additional info:  manually starting a build by submitting a new prow job via curl results in only a single instance being started (however, the build # is out of sync with the build numbers for builds started with `jx start pipeline`.  For example, if `jx start pipeline` will start build `#9,` submitting a prow job will result in a build # of `6`)

**Comment by mikecirioli** [comment]
Does not appear to be originating with the `jx` command, i suspect there is something going on in prow, and am investigating that.

**Comment by rawlingsj** [comment]
Yes recreated when using multiple Jenkins X teams on a single cluster.  Knative build OOTB is using a `ClusterRole`, wonder if we can restrict it to namespace scoped `Role`.  Also worth checking upstream issues.

/kind bug
/priority important-soon
/assign rawlingsj
/area build
